samples/ordered_puzzle/components/ordered_puzzle.js

```diff
@@ -0,0 +1,94 @@
+AFRAME.registerComponent("ordered-puzzle", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    const numPuzzlePieces = data.images.length;
+
+    for (i = 0; i < numPuzzlePieces; i++) {
+      var textBoxProp =
```

Comment:
Instead of manually creating the string version of the JSON object here, you should just be able to do:
```
JSON.stringify(data.images[i])
```
and pass it directly like:
```
this.target.setAttribute("target-box", "jsonData", JSON.stringify(data.images[i]));
```
Note, if you want to add something like color, you will have to do it before stringifying it, like
```
data.images[i].color = "yellow"
```